src/protocols/echo/Echo.h

```diff
@@ -36,11 +36,15 @@
 
 namespace chip {
 namespace Protocols {
+namespace Echo {
 
-enum
+/**
+ * Echo Protocol Message Types
+ */
+enum class MsgType : uint8_t
```

Comment:
enum class are supposed to provide type safety. If we static_cast everywhere that typesafety seems lost.

How about a helper function like:

```
uint8_t  asSendMessageType(MsgType type) {...}
```

and use that instead of static cast?
I am still unsure about API usefulness though - I believe the intent would be to ensure that Echo MsgType can use if and only if protocol is echo, so we would need some way to tie together a protocol + message type in a type safe manner.

Reply:
Yes, we discussed the pro and con as we changed the enum to enum class in SecureChannelProtocol, the pro of this change is this prevent enums class from exporting their enumerators to the surrounding scope, causing name clashes. The con is we need static_cast everywhere.

Helper function seems even more expensive solution.